common/src/math.rs

```diff
@@ -152,6 +152,20 @@ pub fn project_to_plane<N: RealField + Copy>(
         * ((distance - subject.dot(normal)) / projection_direction.dot(normal));
 }
 
+/// Returns the UnitQuaternion that rotates the `from` vector to the `to` vector, or `None` if
+/// `from` and `to` face opposite directions such that their sum has norm less than `epsilon`.
+/// This version is more numerically stable than nalgebra's equivalent function.
```

Comment:
Should we push this upstream?

Reply:
I've thought about that. It's probably worth at least creating an issue for it, although I'll probably want to find more evidence that it's better. The current implementation calls into `scaled_rotation_between_axis`, which I don't think has an equivalent optimization, so this optimization would add a little bit of extra complexity to nalgebra.

Reply:
I filed https://github.com/dimforge/nalgebra/issues/1262 so that this change can hopefully be pushed upstream one way or another. This issue also has a concrete example of the kinds of improvements we see.

Without this new implementation, I was getting noticeable jitter in the player camera orientation.